components/data_stores/actions/get-record-keys/get-record-keys.mjs

```diff
@@ -0,0 +1,114 @@
+  version: "0.0.1",
+  type: "action",
+  props: {
+    app,
+    dataStore: {
+      propDefinition: [
+        app,
+        "dataStore",
+      ],
+    },
+    query: {
+      type: "string",
+      label: "Query",
+      description: "The query to filter the records.",
+    },
+    contains: {
+      type: "boolean",
+      label: "Contains",
+      description: "If true, the query will be used to filter the records that contains the query. If false, the query will be used to filter the records that are equal to the query.",
+      optional: true,
+    },
+    caseInsensitive: {
+      type: "boolean",
+      label: "Case Insensitive",
+      description: "if true, the query will be used to filter the records based on case-Insensitivity, if false, the query will be used to filter the records that are equal to the query.",
+      optional: true,
+    },
+  },
+  methods: {
+    getAllKeysWithRecord(
+      obj,
+      query,
+      contains = false,
+      caseInsensitive = false,
+      path = "",
+    ) {
+      const filteredKeys = [];
+
+      const helper = (obj, path) => {
+        if (Array.isArray(obj)) {
+          for (let i = 0; i < obj.length; i++) {
+            helper(obj[i], path + i + ".");
+          }
+          return;
+        }
+        else if (typeof obj === "object" && obj !== null) {
+          for (let key in obj) {
+            helper(obj[key], path + key + ".");
+          }
+          return;
+        }
+
+        if (!contains && !caseInsensitive && obj == query) {
+          filteredKeys.push(path.slice(0, -1));
+          return;
+        }
+
+        if (contains && caseInsensitive && String(obj).toLowerCase()
+          .includes(query.toLowerCase())
+        ) {
+          filteredKeys.push(path.slice(0, -1));
+          return;
+        }
+
+        if (contains && !caseInsensitive && String(obj).includes(query)) {
+          filteredKeys.push(path.slice(0, -1));
+          return;
+        }
+
+        if (!contains && caseInsensitive
+          && String(obj).toLocaleLowerCase() == query.toLowerCase()
+        ) {
+          filteredKeys.push(path.slice(0, -1));
+          return;
+        }
+      };
+
+      helper(obj, path);
+      return filteredKeys;
+    },
+  },
+  async run({ $ }) {
+    const { dataStore } = this;
+    const keys = await dataStore.keys();
+    const promises = [];
+    for (const key of keys) {
+      promises.push(dataStore.get(key));
+    }
+    const arrData = await Promise.all(promises);
+    const objData = {};
+    for (let i = 0; i < keys.length; i++) {
+      objData[keys[i]] = arrData[i];
+    }
+
+    const filteredKeys = this.getAllKeysWithRecord(
+      objData,
+      this.query,
+      this.contains,
+      this.caseInsensitive,
+    );
+
+    $.export("$summary", `Successfully returned ${filteredKeys.length} key(s).`);
+    return {
+      hasRegisteredKeys: filteredKeys.length > 0,
+      keys: filteredKeys,
```

Comment:
Would it be possible for us to consider returning the values as well?
Since using `contains` may result in values that are different from the query, it might be beneficial to include them.